syntax = "proto3";

package pennsieve;

message Event {
  string source = 1;
  uint64 pageStart = 2;
  uint64 pageEnd = 3;
  double samplePeriod = 4;
  uint64 pointsPerEvent = 5;
  bool isResampled = 6;
  repeated double data = 7;
  repeated uint64 times = 8;
  repeated uint32 spikeGroup = 9;
}

message Instruction {
  string command = 1;
  string argument = 2;
}

message Segment {
  uint64 startTs = 1;
  string source = 2;
  uint64 lastUsed = 3;
  string unit = 4;
  double samplePeriod = 5;
  double requestedSamplePeriod = 6;
  uint64 pageStart = 7;
  bool isMinMax = 8;
  uint64 unitM = 9;
  string segmentType = 10;
  uint64 nrPoints = 11;
  repeated double data = 12;
  uint64 pageEnd = 13;
  string channelName = 14;
}

message IngestSegment {
  string channelId = 1;
  uint64 startTime = 2;
  double samplePeriod = 3;
  repeated double data = 4;
}

message TimeSeriesMessage {
  Segment segment = 3;
  repeated Event event = 4;
  Instruction instruction = 5;
  IngestSegment ingestSegment = 6;
  uint64 totalResponses = 7;
  uint64 responseSequenceId = 8;
}
